test/extended/openstack/egressip.go

```diff
@@ -0,0 +1,475 @@
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(waitOk).To(o.BeTrue(), "Not found the expected assigned node '%s' in '%s' CloudPrivateIPConfig", primaryWorker.Name, egressIPAddrStr)
+		e2e.Logf("Found the expected assigned node '%s' in '%s' CloudPrivateIPConfig", primaryWorker.Name, egressIPAddrStr)
+
+		// Label the secondary worker node as EgressIP assignable node
+		g.By(fmt.Sprintf("Labeling the secondary node '%s' with '%s'", secondaryWorker.Name, egressAssignableLabelKey))
+		e2e.AddOrUpdateLabelOnNode(clientSet, secondaryWorker.Name, egressAssignableLabelKey, "dummy")
+		defer e2e.RemoveLabelOffNode(clientSet, secondaryWorker.Name, egressAssignableLabelKey)
+
+		// Check the assigned node in CloudPrivateIPConfig is kept with the primary EgressIP worker node
+		g.By("Checking the CloudPrivateIPConfig object and the assigned node after adding a second EgressIP assignable node")
+		cpicAssignedNode, err := getCpipAssignedNode(cloudNetworkClientset, egressIPAddrStr)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Could not find the assigned node in '%s' CloudPrivateIPConfig", egressIPAddrStr)
+		e2e.Logf("'%s' CloudPrivateIPConfig assigned node ('%s') is kept after adding a second assignable node", egressIPAddrStr, cpicAssignedNode)
+		o.Expect(cpicAssignedNode).To(o.Equal(primaryWorker.Name), "'%s' egressIP has been deassigned to the '%s' node", egressIPAddrStr, primaryWorker.Name)
+
+		g.By("Checking a port has been created in Openstack for the EgressIP")
+		ports, err := getPortsByIP(networkClient, egressIPAddrStr, machineNetworkID)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Could not find the Openstack port with IP address '%s'", egressIPAddrStr)
+		o.Expect(ports).To(o.HaveLen(1), "Unexpected number of Openstack ports (%d) found with IP address '%s'", len(ports), egressIPAddrStr)
+		egressIPPort := ports[0]
+		e2e.Logf("Found '%s' Openstack port with IP address '%s'", egressIPPort.Name, egressIPAddrStr)
+
+		g.By("Checking that the allowed_addresses_pairs are properly updated for the workers in the Openstack before failover")
+		checkAllowedAddressesPairs(networkClient, primaryWorker, secondaryWorker, egressIPAddrStr, machineNetworkID)
+
+		g.By("Creating a FIP in Openstack")
+		var fip *floatingips.FloatingIP
+		externalNetworkId, err := GetFloatingNetworkID(networkClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		fip, err = floatingips.Create(networkClient, floatingips.CreateOpts{FloatingNetworkID: externalNetworkId}).Extract()
+		o.Expect(err).NotTo(o.HaveOccurred(), "Error creating FIP using discovered floatingNetwork ID '%s'", externalNetworkId)
+		e2e.Logf("The FIP '%s' has been created in Openstack", fip.FloatingIP)
+		defer floatingips.Delete(networkClient, fip.ID)
+
+		g.By(fmt.Sprintf("Attaching the FIP %s to the EgressIP port '%s'", fip.FloatingIP, egressIPPort.Name))
+		floatingips.Update(networkClient, fip.ID, floatingips.UpdateOpts{PortID: &egressIPPort.ID})
```

Comment:
should we also .Extract() here so it returns the obj and err?
consequently we could check if there was an err and there would be no need to do lines 184-188

Reply:
This is a good point!

Reply:
Done, thanks!